elastic/kibana
# [Response Ops][Alerting] Updating `AlertsClient` to provide feature parity with rule registry lifecycle executor

Resolves https://github.com/elastic/kibana/issues/160173

## Summary
The rule registry lifecycle executor automatically sets the following fields in alert docs:
- `event.action` - `open`, `active` or `close` depending on what type of alert
- `event.kind` - always `signal`
- `tags` - merges rule tags with rule executor reported tags
- `kibana.version`
- `kibana.alert.workflow_status` - set to `open`
- `kibana.alert.time_range`

In addition, the rule registry lifecycle executor provides some helper functions for the rule executors to get the alert UUID, the alert start time (if it exists) and the alert document for recovered alerts (used to set recovered context variables).

This PR augments the framework `AlertsClient` to set the same fields and to provide the same functionality to the rule executors. When an alert is reported via the `AlertsClient`, the UUID (either existing or newly generated) and the start time (for ongoing alerts) is returned back to the rule executors. When an executor requests the recovered alerts in order to set context information, the existing alert document is returned.

## To Verify

Check out [this commit](https://github.com/elastic/kibana/pull/160466/commits/dc5bebab1b86c7065fb9ef66b8d01da5491ae5d7) which removes the metric threshold rule from the rule registry lifecycle executor and onboards it to use the framework alerts client. Create a metric threshold rule that creates active alerts and recovers alerts. Inspect the alerts documents to make sure all the fields mentioned above exist. Compare these documents with alerts created using the lifecycle executor.
